monkey/infection_monkey/exploit/tools/utils.py

```diff
@@ -0,0 +1,8 @@
+from typing import Sequence
+
+from infection_monkey.i_puppet import TargetHost
+
+
+def should_attempt_exploit(host: TargetHost, exploitation_ports: Sequence[int]) -> bool:
+    closed_tcp_ports = host.ports_status.tcp_ports.closed
+    return not all([p in closed_tcp_ports for p in exploitation_ports])
```

Comment:
I suggest we invert the logic:
```suggestion
def all_exploitation_ports_are_closed(host: TargetHost, exploitation_ports: Sequence[int]) -> bool:
    closed_tcp_ports = host.ports_status.tcp_ports.closed
    return all([p in closed_tcp_ports for p in exploitation_ports])
```

The callers will also need to drop the `not` from their `if` checks.